ctapipe/irf/select.py

```diff
@@ -0,0 +1,205 @@
+import astropy.units as u
+import numpy as np
+from astropy.table import QTable, vstack
+from pyirf.simulations import SimulatedEventsInfo
+from pyirf.spectral import PowerLaw, calculate_event_weights
+from pyirf.utils import calculate_source_fov_offset, calculate_theta
+
+from ..core import Component, Provenance, QualityQuery
+from ..core.traits import List, Unicode
+from ..io import TableLoader
+
+
+class EventSelector(Component):
+    def __init__(self, event_pre_processor, kind, file, target_spectrum, **kwargs):
+        super().__init__(**kwargs)
+
+        self.epp = event_pre_processor
```

Comment:
Why is `EventPreprocessor` given as positional argument here? It should be a subcomponent configured via the config system.

Also the naming here is a bit strange, `EventPreprocessor` inherits from the quality query, so it makes selections and this `EventSelector` uses a `TableLoader` to read things...